test/PixelRendr/generatePaletteFromRawData.ts

```diff
@@ -0,0 +1,35 @@
+/// <reference path="../../node_modules/@types/chai/index.d.ts" />
+/// <reference path="../../node_modules/@types/mocha/index.d.ts" />
+/// <reference path="../../lib/PixelRendr.d.ts" />
+/// <reference path="../utils/MochaLoader.ts" />
+/// <reference path="../utils/mocks.ts" />
+
+mochaLoader.addTest("throws an error if the given array is not a multiple of 4", (): void => {
```

Comment:
Don't test for these error conditions. It intentionally doesn't check the length of inputs because that would be a slight performance degradation; instead, we just assume the Uint8ClampedArray is valid.